packages/ses/src/ses.d.ts

```diff
@@ -0,0 +1,38 @@
+type safetyPreference = 'safe' | 'unsafe';
+
+type LockdownOpts = {
+  fringeSet?: {
+    add?: Function,
+    has?: Function,
+  },
+  dateTaming?: safetyPreference,
+  errorTaming?: safetyPreference,
+  mathTaming?: safetyPreference,
+  regExpTaming?: safetyPreference,
+
+  [extraOption:string]?: any,
+};
+
+declare var lockdown = (LockdownOpts) => void;
+
+declare var harden = (any) => any;
```

Comment:
Does TS support type parameterization like:?

```ts
declare var harden<T> = T => T;
```

Reply:
I was doing something similar to this, and I do know this syntax you've shown is a great way to write abstract types, but I was finding that this approach here would require each declared instance of harden to have a type? I don't believe the type is set on the result as a result of the parameter, instead both expect definition at the `harden` variable assignment.

There might be a way to do what you've described, but I don't know it, and I don't think it's what you've written there.

Reply:
Example:
```javascript
const aHarden: harden<string> = harden;
// Now we have a string-only version of harden.
```

Reply:
I see. I’ve been spoilt by [Lobster]’s many many type inference mechanisms.

[Lobster]: http://strlen.com/lobster/

Reply:
ts can definitely do generic stuff like

```js
function harden<T>(x: T): T
```

Reply:
oh... I see this was addressed already. nvm